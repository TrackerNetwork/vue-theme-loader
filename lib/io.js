const fs = require('fs');

module.exports = {
  readFile: file =>
    new Promise((resolve, reject) => {
      fs.readFile(file, 'utf8', (err, contents) => {
        if (err) reject(err);
        resolve(contents);
      });
    }),

  writeFile: (file, content) =>
    new Promise((resolve, reject) => {
      fs.writeFile(file, content, err => {
        if (err) reject(err);
        resolve();
      });
    })
}
