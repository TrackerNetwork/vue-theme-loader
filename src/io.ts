import { readFile as read, writeFile as write } from 'fs';

/**
 * Simple promise wrapper around node's readFile
 */
export const readFile = (file: string): Promise<string> =>
  new Promise((resolve, reject) => {
    read(file, 'utf8', (err, contents) => {
      err
        ? reject(err)
        : resolve(contents);
    });
  });

/**
 * Simple promise wrapper around node's writeFile
 */
export const writeFile = (file: string, content: string) =>
  new Promise((resolve, reject) => {
    write(file, content, err => {
      err
        ? reject(err)
        : resolve();
    });
  });
