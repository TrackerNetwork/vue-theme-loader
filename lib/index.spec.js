import { readFile } from './io';
import {
  parse,
  removeOtherThemes,
  removeStyleBlock
} from './index';

test('removeStyleBlock()', async () => {
  const source = await readFile('./fixtures/normalComponent.vue');
  expect(removeStyleBlock(source, parse(source).styles[0])).toMatchSnapshot()
});

test('removeOtherThemes() not modify anything if theme undefined', async () => {
  const source = await readFile('./fixtures/normalComponent.vue');
  expect(removeOtherThemes(source)).toMatchSnapshot();
});

test('removeOtherThemes() should remove unmatching theme', async () => {
  const source = await readFile('./fixtures/themed.vue');
  expect(removeOtherThemes(source, 'brand1')).toMatchSnapshot();
});
