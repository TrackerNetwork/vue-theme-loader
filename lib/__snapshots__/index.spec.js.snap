exports[`test removeOtherThemes() not modify anything if theme undefined 1`] = `
"<template test=\"213\">
  .Component
</template>

<script abc=\"123\">
export default {
}
</script>

<style test=\"true\">
  .Component {
    color: red;
  }
</style>

<style></style>
"
`;

exports[`test removeOtherThemes() should remove unmatching theme 1`] = `
"<style>
  .doNotTouch {
    color: red;
  }
</style>












"
`;

exports[`test removeStyleBlock() 1`] = `
"<template test=\"213\">
  .Component
</template>

<script abc=\"123\">
export default {
}
</script>







<style></style>
"
`;
